components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/scehduler/CoordinatedTaskScheduler.java

```diff
@@ -89,12 +94,44 @@ public void run() {
                 LOG.debug("This node is not leader. Hence not cleaning task store or resolving un assigned tasks.");
             }
             // schedule all tasks assigned to this node and in state none
-            scheduleAllTasksAssignedToThisNode();
+            scheduleAssignedTasks(CoordinatedTask.States.NONE);
         } catch (Throwable throwable) { // catching throwable to prohibit permanent stopping of the executor service.
             LOG.fatal("Unexpected error occurred while trying to schedule tasks.", throwable);
         }
     }
 
+    /**
+     * Pause ( stop execution ) the deactivated tasks.
+     *
+     * @throws TaskCoordinationException - when something goes wrong while retrieving tasks information from store.
+     */
+    private void pauseDeactivatedTasks() throws TaskCoordinationException {
+
+        List<String> deactivatedTasks = taskStore.retrieveTaskNames(localNodeId, CoordinatedTask.States.DEACTIVATED);
+        if (LOG.isDebugEnabled()) {
+            deactivatedTasks.stream().map(
+                    task -> "Task [" + task + "] retrieved in [" + CoordinatedTask.States.DEACTIVATED + "] state.")
+                    .forEachOrdered(LOG::debug);
+        }
+        ScheduledTaskManager scheduledTaskManager = taskManager;
+        List<String> pausedTasks = new ArrayList<>();
+        List<String> deployedTasks = taskManager.getAllCoordinatedTasksDeployed();
+        deactivatedTasks.forEach(task -> {
+            if (deployedTasks.contains(task)) {
+                try {
+                    scheduledTaskManager.stopExecution(task);
+                    pausedTasks.add(task);
+                } catch (TaskException e) {
+                    LOG.error("Error stopping the task [" + task + "]", e);
```

Comment:
If we get an error here, the exception will be logged and ignored, but the state in the DB will get updated to PAUSED, is that ok?

Reply:
It will not get updated, since we are adding the list of tasks which are paused successfully to the list **pausedTasks.add(task);** and then only updating them to db in line 132. If it is an exception, it won't get added to the list.